code/database/driver/abstract.php

```diff
@@ -196,10 +196,20 @@ public function disconnect()
      * Provides access to the underlying database connection. Useful for when
      * you need to call a proprietary method such as postgresql's lo_* methods
      *
+     * @param bool $auto_connect If connection hasn't been established, auto connect
      * @return resource
+     * @throws \RuntimeException if auto connection fails
      */
-    public function getConnection()
+    public function getConnection($auto_connect = true)
     {
+        if (!$this->_connection && $auto_connect) {
```

Comment:
@oligriffiths Should use isConnected() here instead of this->_connection, same below.